src/stores/ShardDetail.ts

```diff
@@ -24,15 +40,107 @@ export interface ShardDetails {
+    getShardStatusMessageId: (shardId: string) => ShardStatusMessageIds;
+    evaluateShardProcessingState: (shardId: string) => boolean;
 }
 
-const defaultStatusColor = slate[25];
+export const defaultStatusColor: ShardStatusColor = '#EEF8FF';
+
+const defaultTaskShardDetail: TaskShardDetails = {
+    messageId: ShardStatusMessageIds.NONE,
+    color: defaultStatusColor,
+};
 
-const evaluateShardStatus = ({ status }: Shard): string => {
+const evaluateTaskShardStatus = ({ spec, status }: Shard): TaskShardDetails => {
+    if (status.length === 1) {
+        switch (status[0].code) {
+            case 'FAILED':
+                return {
+                    messageId: ShardStatusMessageIds.FAILED,
+                    color: errorMain,
+                };
+            case 'PRIMARY':
+                return {
+                    messageId: ShardStatusMessageIds.PRIMARY,
+                    color: successMain,
+                };
+            case 'IDLE':
+                return {
+                    messageId: ShardStatusMessageIds.IDLE,
+                    color: warningMain,
+                };
+            case 'STANDBY':
+                return {
+                    messageId: ShardStatusMessageIds.STANDBY,
+                    color: warningMain,
+                };
+            case 'BACKFILL':
+                return {
+                    messageId: ShardStatusMessageIds.BACKFILL,
+                    color: warningMain,
+                };
+            default:
+                return {
+                    messageId: ShardStatusMessageIds.NONE,
+                    color: defaultStatusColor,
+                };
+        }
+    } else if (status.length > 1) {
+        const statusCodes: (ReplicaStatusCode | undefined)[] = status.map(
+            ({ code }) => code
+        );
+
+        if (statusCodes.find((code) => code === 'FAILED')) {
+            return {
+                messageId: ShardStatusMessageIds.FAILED,
+                color: errorMain,
+            };
+        } else if (statusCodes.find((code) => code === 'PRIMARY')) {
+            return {
+                messageId: ShardStatusMessageIds.PRIMARY,
+                color: successMain,
+            };
+        } else if (statusCodes.find((code) => code === 'IDLE')) {
+            return {
+                messageId: ShardStatusMessageIds.IDLE,
+                color: warningMain,
+            };
+        } else if (statusCodes.find((code) => code === 'STANDBY')) {
+            return {
+                messageId: ShardStatusMessageIds.STANDBY,
+                color: warningMain,
+            };
+        } else if (statusCodes.find((code) => code === 'BACKFILL')) {
+            return {
+                messageId: ShardStatusMessageIds.BACKFILL,
+                color: warningMain,
+            };
+        } else {
+            return {
+                messageId: ShardStatusMessageIds.NONE,
+                color: defaultStatusColor,
+            };
+        }
```

Comment:
Agree with the TODO... Not with this PR - but this is a good place to refactor. As this is the same logic duplicated and slightly changed. We should make this a single approach and if there is a single status just pop it into an array so we can use the `find` approach in both.